### PR TITLE
Add request parameter 'cluster_manager_timeout' and deprecate 'master_timeout' - in Ingest APIs and Script APIs

### PR DESCRIPTION
**Pull request by tlfeng**
### Description

- Deprecate the request parameter `master_timeout` that used in Ingest APIs and Script APIs which have got the parameter. 
- Add alternative new request parameter `cluster_manager_timeout`.
- Add unit tests.

List of the `Ingest` APIs and `Script` APIs that have got request parameter `master_timeout`:
- In Ingest API:
Put pipeline API https://opensearch.org/docs/latest/opensearch/rest-api/ingest-apis/create-update-ingest/
Get pipeline API https://opensearch.org/docs/latest/opensearch/rest-api/ingest-apis/get-ingest/
Delete pipeline API https://opensearch.org/docs/latest/opensearch/rest-api/ingest-apis/delete-ingest/
- In Script API:
Get stored script API `GET _scripts/<script-id>`
Put stored script API
Delete stored script API

### Issues Resolved
A part of issue https://github.com/opensearch-project/OpenSearch/issues/2511
 
### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tlfeng** [comment]
In log 3964:
```
REPRODUCE WITH: ./gradlew ':client:rest-high-level:asyncIntegTest' --tests "org.opensearch.client.StoredScriptsIT.testDeleteStoredScript" -Dtests.seed=555C66C872D25111 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es -Dtests.timezone=Israel -Druntime.java=17

org.opensearch.client.StoredScriptsIT > testDeleteStoredScript FAILED
    org.opensearch.client.WarningFailureException: method [DELETE], host [http://[::1]:36429], URI [/_scripts/calculate-score?master_timeout=50s&timeout=50s], status line [HTTP/1.1 200 OK]
    Warnings: [Deprecated parameter [master_timeout] used. To promote inclusive language, please use [cluster_manager_timeout] instead. It will be unsupported in a future major version.]
    {"acknowledged":true}
        at __randomizedtesting.SeedInfo.seed([555C66C872D25111:D1CB1B2F4ED7FB98]:0)
        at app//org.opensearch.client.RestClient.convertResponse(RestClient.java:346)
        at app//org.opensearch.client.RestClient$1.completed(RestClient.java:400)
        at app//org.opensearch.client.RestClient$1.completed(RestClient.java:396)
        at app//org.apache.http.concurrent.BasicFuture.completed(BasicFuture.java:122)
        at app//org.apache.http.impl.nio.client.DefaultClientExchangeHandlerImpl.responseCompleted(DefaultClientExchangeHandlerImpl.java:181)
```

**Comment by tlfeng** [comment]
In log 4063:
```
> Task :client:rest-high-level:asyncIntegTest

REPRODUCE WITH: ./gradlew ':client:rest-high-level:asyncIntegTest' --tests "org.opensearch.client.BulkProcessorIT.testGlobalParametersAndBulkProcessor" -Dtests.seed=F88723257C886B84 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es-ES -Dtests.timezone=Asia/Pyongyang -Druntime.java=17

org.opensearch.client.BulkProcessorIT > testGlobalParametersAndBulkProcessor FAILED
    org.opensearch.client.WarningFailureException: method [PUT], host [http://127.0.0.1:38971], URI [/_ingest/pipeline/pipeline_id?master_timeout=30s&timeout=30s], status line [HTTP/1.1 200 OK]
    Warnings: [Deprecated parameter [master_timeout] used. To promote inclusive language, please use [cluster_manager_timeout] instead. It will be unsupported in a future major version.]
    {"acknowledged":true}
```
The Client test failure will be resolved in https://github.com/opensearch-project/OpenSearch/pull/2702

**Review by dreamer-89 — APPROVED** [review]
Nice, LGTM!

**Comment by tlfeng** [comment]
In log 4128:
```
REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v2.0.0#mixedClusterTest' --tests "org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=tasks.list/10_basic/tasks_list test}" -Dtests.seed=E160957F1503B0FE -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ar-MA -Dtests.timezone=Europe/Ulyanovsk -Druntime.java=17

org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT > test {p0=tasks.list/10_basic/tasks_list test} FAILED
    java.lang.AssertionError: Failure at [tasks.list/10_basic:15]: field [nodes.$node_id.roles] doesn't have a true value
    Expected: not null
         but: was null
        at __randomizedtesting.SeedInfo.seed([E160957F1503B0FE:6934AAA5BBFFDD06]:0)
        at org.opensearch.test.rest.yaml.OpenSearchClientYamlSuiteTestCase.executeSection(OpenSearchClientYamlSuiteTestCase.java:454)
        at org.opensearch.test.rest.yaml.OpenSearchClientYamlSuiteTestCase.test(OpenSearchClientYamlSuiteTestCase.java:427)
```

**Comment by tlfeng** [comment]
In log 4156:
```
> Task :server:test

REPRODUCE WITH: ./gradlew ':server:test' --tests "org.opensearch.cluster.coordination.JoinTaskExecutorTests.testPreventJoinClusterWithUnsupportedNodeVersions" -Dtests.seed=4F91E1AE4511AA5E -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=mt -Dtests.timezone=Europe/Belfast -Druntime.java=17

org.opensearch.cluster.coordination.JoinTaskExecutorTests > testPreventJoinClusterWithUnsupportedNodeVersions FAILED
    junit.framework.AssertionFailedError: Expected exception IllegalStateException but no exception was thrown
        at __randomizedtesting.SeedInfo.seed([4F91E1AE4511AA5E:CCEE46D497D23A12]:0)
        at org.apache.lucene.tests.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2898)
        at org.apache.lucene.tests.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2884)
        at org.opensearch.cluster.coordination.JoinTaskExecutorTests.testPreventJoinClusterWithUnsupportedNodeVersions(JoinTaskExecutorTests.java:122)
```
Also seen in #2752

**Comment by tlfeng** [comment]
In log 4161:
```
REPRODUCE WITH: ./gradlew ':qa:remote-clusters:integTest' --tests "org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks" -Dtests.seed=C9D4052D39E3B73 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ru -Dtests.timezone=America/Cancun -Druntime.java=17

org.opensearch.cluster.remote.test.RemoteClustersIT > testHAProxyModeConnectionWorks FAILED
    java.lang.AssertionError
        at __randomizedtesting.SeedInfo.seed([C9D4052D39E3B73:BEB9599E8184D2E]:0)
        at org.junit.Assert.fail(Assert.java:87)
        at org.junit.Assert.assertTrue(Assert.java:42)
        at org.junit.Assert.assertTrue(Assert.java:53)
        at org.opensearch.cluster.remote.test.RemoteClustersIT.testHAProxyModeConnectionWorks(RemoteClustersIT.java:125)
```
It's reported in issue #1703

File: rest-api-spec/src/main/resources/rest-api-spec/api/delete_script.json [hunk 1]
```diff
@@ -28,7 +28,15 @@
       },
       "master_timeout":{
         "type":"time",
-        "description":"Specify timeout for connection to master"
+        "description":"Specify timeout for connection to master",
+        "deprecated":{
+          "version":"2.0.0",
+          "description":"To promote inclusive language, use 'cluster_manager_timeout' instead."
```
Comment (anasalkouz) [inline rest-api-spec/src/main/resources/rest-api-spec/api/delete_script.json:34]: It feels to me promote is the not the right work. may be to support inclusive language. or to be more inclusive, use ..

Comment (tlfeng) [reply rest-api-spec/src/main/resources/rest-api-spec/api/delete_script.json:34]: Thanks for your opinion! I will change the word `promote` to `support`.

Comment (tlfeng) [reply rest-api-spec/src/main/resources/rest-api-spec/api/delete_script.json]: Changed in commit https://github.com/opensearch-project/OpenSearch/pull/2682/commits/7d0de096a0b1213ae9ba39bfa88623b699e92973

**Review by xuezhou25 — APPROVED** [review]

**Pull request merged**